Client/game_sa/HookSystem.h

```diff
@@ -16,7 +16,33 @@
 
 #define     MAX_JUMPCODE_SIZE           20
 
-BOOL  HookInstall(DWORD dwInstallAddress, DWORD dwHookHandler, int iJmpCodeSize);
+template <typename T>
+void* FunctionPointerToVoidP(T func)
+{
+    union
+    {
+        T     a;
+        void* b;
+    } c = {func};
+    return c.b;
+}
+
+template <typename T>
+BOOL HookInstall(DWORD dwInstallAddress, T dwHookHandler, int iJmpCodeSize = 5)
+{
+    BYTE JumpBytes[MAX_JUMPCODE_SIZE];
+    MemSetFast(JumpBytes, 0x90, MAX_JUMPCODE_SIZE);
+    if (CreateJump(dwInstallAddress, (DWORD)FunctionPointerToVoidP(dwHookHandler), JumpBytes))
+    {
+        MemCpy((PVOID)dwInstallAddress, JumpBytes, iJmpCodeSize);
+        return TRUE;
+    }
+    else
+    {
+        return FALSE;
+    }
+}
+
```

Comment:
Correct me if I am wrong, but this seems like a pretty core change that isn't related to this crash fix. Is there a particular reason it's here instead of in a separate PR?

Reply:
The change is needed by this PR because, in C++, you can't get the address of class methods and cast it to unsigned int or DWORD, so this is basically a dirty hack that allows us to get the address of class methods. If I remove this change, then this code will not compile:
```
HookInstall(0x534120, &CEntitySAInterface::GetBoundRect_);
```